### PR TITLE
pull in changes from oasis-core master

### LINKED ISSUE #9: Update ts-web for oasis-core master branch
tasks breakdown forthcoming

but we'll be pursuing this mainly for its expanded runtime interfaces

### PR DESCRIPTION
**Pull request by pro-wh**
fixes #9 

This brings ts-web/core to oasis-core rev e000dfaf06a1e3a58f1eb3ca99c24922fac7aea5

### CODE REVIEW

File: client-sdk/ts-web/core/src/beacon.ts [hunk 1]
```diff
@@ -1,3 +1,24 @@
...
+ */
+export const METHOD_PVSS_COMMIT = 'beacon.PVSSCommit';
+/**
+ * MethodPVSSReveal is the method name for a PVSS reveal.
+ */
+export const METHOD_PVSS_REVEAL = 'beacon.PVSSReveal';
+/**
+ * MethodSetEpoch is the method name for setting epochs.
+ */
+export const METHOD_SET_EPOCH = '000_beacon.SetEpoch';
```
Comment (kostko) [inline client-sdk/ts-web/core/src/beacon.ts:12]: Not sure if these need to be exposed in the client SDK as the first two can only be called by a validator and the last one can only be used in debug mode.

File: client-sdk/ts-web/core/src/consensus.ts [hunk 2]
```diff
@@ -78,6 +82,8 @@ export const CODE_INSUFFICIENT_FEE_BALANCE = 2;
  */
 export const CODE_GAS_PRICE_TOO_LOW = 3;
 
+// %%% method priority?
```
Comment (kostko) [inline client-sdk/ts-web/core/src/consensus.ts:85]: I don't think method priority needs to be exposed in the client SDK as there is nothing there that the clients can do specifically.

File: client-sdk/ts-web/core/src/types.ts [hunk 3]
```diff
@@ -1208,6 +1478,12 @@ export interface RegistryUnfreezeNode {
     node_id: Uint8Array;
 }
 
+/**
+ * ValidatorSetConstraint specifies that the entity must have a node that is part of the validator
+ * set. No other options can currently be specified.
+ */
+export type RegistryValidatorSetConstraint = Map<never, never>;
```
Comment (kostko) [inline client-sdk/ts-web/core/src/types.ts:1485]: The constraints types should probably be grouped together? Now it seems that some are 300 lines higher.

Comment (kostko) [reply client-sdk/ts-web/core/src/types.ts:1485]: Also why does it have this `Registry` prefix that the original type does not have?

File: client-sdk/ts-web/core/src/beacon.ts (hunk 1 above)
Comment (pro-wh) [reply client-sdk/ts-web/core/src/beacon.ts:12]: I'm fine with exposing methods used in debug mode.

File: client-sdk/ts-web/core/src/consensus.ts (hunk 2 above)
Comment (pro-wh) [reply client-sdk/ts-web/core/src/consensus.ts]: yeah agreed

File: client-sdk/ts-web/core/src/types.ts (hunk 3 above)
Comment (pro-wh) [reply client-sdk/ts-web/core/src/types.ts:1642]: I went with alphabetical sorting in types.ts ):

these types almost all have a prefix for what Go package they're from

**Comment by pro-wh** [comment]
oh right I was gonna check how the kyber stuff ended up serializing

File: client-sdk/ts-web/core/src/types.ts [hunk 4]
```diff
@@ -96,6 +160,16 @@ export interface CommonConsensusInfo {
     addresses: CommonConsensusAddress[];
 }
 
+/**
+ * Constraints are the Intel SGX TEE constraints.
+ */
+export interface CommonConstraints {
```
Comment (kostko) [inline client-sdk/ts-web/core/src/types.ts:166]: This is originally under `sgx.Constraints` so maybe `SGXConstraints`?

Comment (pro-wh) [reply client-sdk/ts-web/core/src/types.ts:166]: ya there's definitely some weirdness with this being a flat namespace and the Go codebase being hierarchical. currently there's a `Common` type name prefix and a `common.ts` with almost everything from there mixed together: node, entity, sgx, etc. these are ones that are mostly _type definitions_ and _constants_. the exceptions are modules that are mostly _helpers_: common/cbor -> `misc.ts` and common/crypto/... -> their own file and prefix, such as `signature.ts` and `hash.ts`. PVSS is on the fence.

Comment (pro-wh) [reply client-sdk/ts-web/core/src/types.ts:166]: let's change `CommonConstraints` to `SGXConstraints` and similar. `CommonNode` will be `Node`. `SignatureSignature` will become `Signature`. but keep helpers and constants in `common.ts`.

**Review by kostko — APPROVED** [review]

**Pull request merged**